### PR TITLE
Update SentMessage.php

### PR DESCRIPTION
**Pull request by dima-gr**
Sometimes the response from the server can be null, because of this, the application is terminated

| Q             | A
| ------------- | ---
| Branch?       | 5.4 for features / 4.4 or 5.3 for bug fixes 
| Bug fix?      | yes/no
| New feature?  | yes/no 
| Deprecations? | yes/no 
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

File: src/Symfony/Component/Notifier/Message/SentMessage.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ public function getTransport(): string
         return $this->transport;
     }
 
-    public function setMessageId(string $id): void
+    public function setMessageId(?string $id): void
```
Comment (fabpot) [inline src/Symfony/Component/Notifier/Message/SentMessage.php:39]: I would prefer to avoid calls to this method with null if possible.

**Comment by OskarStark** [comment]
Which transport are you using ?

**Comment by dima-gr** [comment]
firebase - Symfony\Component\Notifier\Bridge\Firebase
and when the response comes null, then in the logs appears warning that the unknown index 'message_id', and then a critical error that the parameter cannot be null
`
$sentMessage->setMessageId($success['results'][0]['message_id']);
`
you can try to check for the existence of id in the response, if it does not exist, then pass an empty string to setMessageId. Maybe it will be better

**Comment by OskarStark** [comment]
Yes please check and only call setMessageId in the Firebase transport if it's a string

**Comment by dima-gr** [comment]
i can modify Symfony\Component\Notifier\Bridge\Firebase\FirebaseTransport.php
and check something like that 
`$sentMessage->setMessageId(isset($success['results'][0]['message_id']) ? : '');`
and then make pull request

**Comment by nicolas-grekas** [comment]
I saw that you removed your fork.
Please submit the PR again when you're ready, and please take time to properly fill in the PR template, the commit message, add a test case if possible also.
Thanks.

**Comment by fabpot** [comment]
@dima-gr Your suggestion is indeed the fix that I would like to see.

**Pull request closed without merging**